src/plugins/vis_type_pie/public/editor/components/pie.tsx

```diff
@@ -0,0 +1,222 @@
+ * Side Public License, v 1.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { METRIC_TYPE } from '@kbn/analytics';
+import { EuiPanel, EuiTitle, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import {
+  BasicOptions,
+  SwitchOption,
+  SelectOption,
+  PalettePicker,
+} from '../../../../vis_default_editor/public';
+import { VisEditorOptionsProps } from '../../../../visualizations/public';
+import { TruncateLabelsOption } from './truncate_labels';
+import { PaletteRegistry } from '../../../../charts/public';
+import { PieVisParams, LabelPositions, ValueFormats, PieTypeProps } from '../../types';
+import { getLabelPositions, getValuesFormats } from '../collections';
+import { getLegendPositions } from '../positions';
+
+export interface PieOptionsProps extends VisEditorOptionsProps<PieVisParams>, PieTypeProps {}
+
+const PieOptions = (props: PieOptionsProps) => {
+  const { stateParams, setValue, aggs } = props;
+  const setLabels = <T extends keyof PieVisParams['labels']>(
+    paramName: T,
+    value: PieVisParams['labels'][T]
+  ) => setValue('labels', { ...stateParams.labels, [paramName]: value });
+  const legendUiStateValue = props.uiState?.get('vis.legendOpen');
+  const [palettesRegistry, setPalettesRegistry] = useState<PaletteRegistry | undefined>(undefined);
+  const [legendVisibility, setLegendVisibility] = useState<boolean>(() => {
+    const bwcLegendStateDefault = stateParams.addLegend == null ? false : stateParams.addLegend;
+    return props.uiState?.get('vis.legendOpen', bwcLegendStateDefault) as boolean;
+  });
+  const hasSplitChart = aggs?.aggs?.find((agg) => agg.schema === 'split' && agg.enabled);
+
+  useEffect(() => {
+    setLegendVisibility(legendUiStateValue);
+  }, [legendUiStateValue]);
+
+  useEffect(() => {
+    const fetchPalettes = async () => {
+      const palettes = await props.palettes?.getPalettes();
+      setPalettesRegistry(palettes);
+    };
+    fetchPalettes();
+  }, [props.palettes]);
+
+  return (
+    <>
+      <EuiPanel paddingSize="s">
+        <EuiTitle size="xs">
+          <h3>
+            <FormattedMessage
+              id="visTypePie.editors.pie.pieSettingsTitle"
+              defaultMessage="Pie settings"
+            />
+          </h3>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <SwitchOption
+          label={i18n.translate('visTypePie.editors.pie.donutLabel', {
+            defaultMessage: 'Donut',
+          })}
+          paramName="isDonut"
+          value={stateParams.isDonut}
+          setValue={setValue}
+        />
+        <BasicOptions {...props} legendPositions={getLegendPositions} />
+        {props.showElasticChartsOptions && (
+          <>
+            <SwitchOption
+              label={i18n.translate('visTypePie.editors.pie.distinctColorsLabel', {
+                defaultMessage: 'Distinct colors per slice',
```

Comment:
This setting is not relevant for single-ring pie charts, only multi-ring. Can it be disabled until it's needed, since it's confusing to have a toggle that doesn't update the UI?

Reply:
This is a very good point! Done!